src/Sphere.java

```diff
@@ -0,0 +1,28 @@
+public class Sphere extends Shape3D {
+    protected double radius;
+
+    public Sphere(String name, String color, Location loc, double radius) {
+        this.name = name;
+        this.color = color;
+        this.loc = loc;
+        this.radius = radius;
+    }
```

Comment:
Instead of this constructor, do this:
```java
public Sphere(String name, String color, Location loc, double radius) {
    super(name, color, loc);
    this.radius = radius;
}
```